Where_Is_My_Wife/Assets/_Main/Scripts/Player/Manager/PlayerManager.cs

```diff
@@ -36,7 +38,9 @@ private void Start()
             _playerInputEvent = InputEventManager.Instance.PlayerInputEvent;
          
             SubscribeToObservables();
-            
+
+            _canDash = true;
```

Comment:
Mmmm this may be breaking the hierarchy, wasn't the state machine the responsible of allowing and disallowing which state can go to which one? @FerchusGames

Reply:
Let me check how it works here

Reply:
> wasn't the state machine the responsible of allowing and disallowing which state can go to which one? 

It still applies with Abraham's implementation. Here, he is just using this variable to decide whether or not to invoke the Dash event, the state machine is still responsible for choosing to listen to the event and transition or not.